ddtrace/tracer/span.go

```diff
@@ -912,6 +913,28 @@ func (s *Span) AddEvent(name string, opts ...SpanEventOption) {
 	s.spanEvents = append(s.spanEvents, event)
 }
 
+// RecordException attaches error information to the current span as a span event.
+// Additional span event attributes can be provided. The values may be (array of)
+// string, boolean, int or float type.
+func (s *Span) RecordException(err error, attributes ...map[string]any) {
+	if err == nil {
+		return
+	}
```

Comment:
What if `err` is nil, but `attributes` contains exception.type/exception.message/exception.stacktrace? 

As the code is written, nothing will get recorded. But what if a user wants to just populate these fields manually to force an exception to be recorded, without an error? 🤔